scripts/linters/pre_commit_linter.py

```diff
@@ -418,29 +418,49 @@ def categorize_files(file_paths):
     _FILES.update(all_filepaths_dict)
 
 
-def _print_complete_summary_of_errors(all_messages):
-    """Print complete summary of errors."""
-    error_messages = all_messages
-    if error_messages != '':
+def _print_complete_summary_of_lint_messages(lint_messages):
+    """Print complete summary of lint messages."""
+    if lint_messages != '':
         python_utils.PRINT('Summary of Errors:')
         python_utils.PRINT('----------------------------------------')
-        for message in error_messages:
+        for message in lint_messages:
             python_utils.PRINT(message)
 
 
-def _get_task_output(all_messages, task, semaphore):
+def _get_task_output(lint_messages, task, semaphore):
     """Returns output of running tasks.
 
     Args:
-        all_messages: list(str). List of summary messages of linter output.
+        lint_messages: list(str). List of summary messages of linter output.
         task: object(TestingTaskSpec). The task object to get output of linter.
         semaphore: threading.Semaphore. The object that controls how many tasks
             can run at any time.
     """
-    all_messages += task.output
+    if task.output:
```

Comment:
This entire function doesn't make sense to me as it starts with `get` and docstring says it "Returns ..." but it does not return anything! [Make sure to cover this in the issue where we are thinking to remove the semaphore from this method]

Reply:
Okay, I will keep that in mind.

Reply:
Can you please add this in the issue thread so that we don't forget?

Reply:
Done.